src/cluster_asm.h

```diff
@@ -13,8 +13,79 @@
 struct asmTask;
 struct asmTaskManager;
 
+void clusterAsmInit(void);
+
 void clusterMigrationCommand(client *c);
 void clusterSyncSlotsCommand(client *c);
 
+
+/* API for implementation/plugin
+ *
+ * - On destination side, implementation calls clusterAsmRequest(ASM_REQUEST_IMPORT_START)
+ *   to start the import operation
+ * - Redis calls clusterAsmOnEvent() when an event occurs.
+ * - On the source side, Redis will call clusterAsmOnEvent(ASM_EVENT_IMPORT_WAIT_PAUSE)
+ *   when the write pause is needed.
+ * - Implementation stops the traffic to the slots and calls clusterAsmRequest(ASM_REQUEST_IMPORT_PAUSED)
+ * - On the destination side, Redis calls clusterAsmOnEvent(ASM_EVENT_IMPORT_COMPLETED)
+ *   when the import is completed.
+ * - Plugin calls clusterAsmRequest(ASM_REQUEST_CONFIG_UPDATED) to notify Redis
+ *   that the config is updated.
+ *
+ * Sequence diagram for import:
+ *   - Note: shows only the events that plugin needs to react.
+ *
+ * ┌───────────────┐              ┌───────────────┐         ┌───────────────┐             ┌───────────────┐
+ * │ Destination   │              │ Destination   │         │    Source     │             │ Source        │
+ * │ Cluster plugin│              │ Master        │         │    Master     │             │ Cluster plugin│
+ * └───────┬───────┘              └───────┬───────┘         └───────┬───────┘             └───────┬───────┘
+ *         │                              │                         │                             │
+ *         │ ASM_REQUEST_IMPORT_START     │                         │                             │
+ *         ├─────────────────────────────►│                         │                             │
+ *         │                              │CLUSTER SYNCSLOTS <arg>  │                             │
+ *         │                              ├────────────────────────►│                             │
+ *         │                              │                         │                             │
+ *         │                              │  SNAPSHOT(restore cmds) │                             │
+ *         │                              │◄────────────────────────┤                             │
+ *         │                              │  Repl stream            │                             │
+ *         │                              │◄────────────────────────┤                             │
+ *         │                              │                         │ ASM_EVENT_IMPORT_WAIT_PAUSE │
```

Comment:
migrate_wait_pause?

Reply:
yes, the pause operation happen on source nodes, so `migrate` word is better. your suggestion
looks good to me